CIPs/CIP-122.md

```diff
@@ -0,0 +1,454 @@
+---
+cip: 122
+title: 3 DID Method
+author: Joel Thorstensson (@oed)
+discussions-to: https://forum.ceramic.network/t/cip-122-3-did-method
+status: Draft
+category: Standards
+type: Core
+created: 2021-02-12
+updated: 2023-01-31
+replaces: 79
+---
+
+## Simple Summary
+
+<!--Provide a simplified and layman-accessible explanation of the CIP.-->
+The 3 DID method enables users to compose multiple accounts into a single identifier.
+
+
+## Abstract
+
+<!--A short (~200 word) description of the technical issue being addressed.-->
+With a special type of Ceramic stream 3ID enables a revocation registry that can be used to add and remove verification methods from the 3 DID method as well as revoke any capability issued by the identifier. The revocation registry is based on hashes of object-capabilities encoded as CACAO.
+
+
+## Motivation
+
+<!--Motivation is critical for CIPs that want to change the Ceramic protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the CIP solves. CIP submissions without sufficient motivation may be rejected outright.-->
+Currently in Ceramic the main accounts types are PKH DID. These are great because they enable existing wallets to be used directly with Ceramic. However, these accounts are inherently tied to a specific account and there is no way to rotate keys in a simple manner. The 3 DID method changes this by defining a DID method which works with object-capabilities and supports all DID CRUD operations.
+
+Furthermore, the revocation regsitry of the 3 DID method enables the DID subject to revoke any object-capability issued by the 3ID. This can be useful in case of a key compromise, e.g. for a temporary session key.
+
+## Specification
+
+<!--The technical specification should describe the syntax and semantics of any new feature.-->
+Specification goes here.
+
+### The did:3 identifier
+
+> ```
+> did:3:<cacao-mh>
+> ```
+
+The `<cacao-mh>` value is a `keccak-256` multihash over the CID of a CACAO object-capability, e.g. ReCap or UCAN. The value MUST be encoded as a multibase string, using `base58btc`.
```

Comment:
Link to ReCap and UCAN?

Reply:
https://eips.ethereum.org/EIPS/eip-5573
https://github.com/ucan-wg/spec